server/src/callbacks.js

```diff
@@ -140,10 +140,38 @@ Empirica.on("batch", async (ctx, { batch }) => {
         .slice(0, 13)}_${config?.batchName}`;
       batch.set("label", batchLabel);
 
-      // set filenames for storing data
-      const scienceDataFilename = `${process.env.DATA_DIR}/batch_${batchLabel}.scienceData.jsonl`;
-      batch.set("scienceDataFilename", scienceDataFilename);
-      fs.closeSync(fs.openSync(scienceDataFilename, "a")); // create an empty datafile
+      // set filenames for storing data
+      const scienceDataFilename = `${process.env.DATA_DIR}/batch_${batchLabel}.scienceData.jsonl`;
+      batch.set("scienceDataFilename", scienceDataFilename);
+      fs.closeSync(fs.openSync(scienceDataFilename, "a")); // create an empty datafile
+      
+      // Validate GitHub repository access before attempting to push data
```

Comment:
Make clear that first we check for read access (because it's fast) and then afterwards check write access by writing a test file.